packages/hierarchies/src/hierarchies/internal/FilteringHierarchyDefinition.ts

```diff
@@ -56,7 +57,9 @@ export class FilteringHierarchyDefinition implements HierarchyDefinition {
         // instance nodes get the auto-expand flag in `parseNode`, but grouping ones need to be handled during post-processing
         HierarchyNode.isClassGroupingNode(node) &&
         node.children.some(
-          (child: ProcessedHierarchyNode) => child.filtering && (child.filtering.isFilterTarget || child.filtering.filteredChildrenIdentifierPaths),
+          (child: ProcessedHierarchyNode) =>
+            child.filtering &&
+            (child.filtering.isFilterTarget || child.filtering.filteredChildrenIdentifierPaths?.some((path) => !("path" in path) || path.options?.autoExpand)),
```

Comment:
IMO, the `HierarchyNodeIdentifiersPath` variant should be equal in behavior as `{ path: HierarchyNodeIdentifiersPath }` and `autoExpand` should default to `false`. I understand that's a breaking change, but IMO it's fine as long as the change is documented, since we're still on `0.x`.